src/Apps/InfiniteDiscovery/InfiniteDiscoveryApp.tsx

```diff
@@ -0,0 +1,310 @@
+import React, { useEffect } from "react"
+import { uniq, sampleSize, shuffle } from "lodash"
+
+const ES_URL =
+  "https://vpc-artsy-staging-es7-srsqgcck5vzy7uq2eqqhvrm4x4.us-east-1.es.amazonaws.com"
+
+const BUILD_SEARCH_QUERY = (lArtworks, dArtworks) => {
+  const likedArtworks = lArtworks.map(artwork => {
+    return {
+      _index: "artworks_7shards_production",
+      _type: "_doc",
+      _id: artwork,
+    }
+  })
+
+  const dismissedArtworks = dArtworks.map(artwork => ({
+    term: {
+      _id: artwork,
+    },
+  }))
+
+  return {
+    query: {
+      bool: {
+        must: [
+          {
+            more_like_this: {
+              fields: ["id", "genes_vectors"],
+              like: likedArtworks,
+              min_term_freq: 1,
+              max_query_terms: 12,
+            },
+          },
+        ],
+        filter: [
+          {
+            term: { published: true },
+          },
+          {
+            term: { for_sale: true },
+          },
+          { range: { career_stage: { gte: 0, lte: 100 } } },
+        ],
+        must_not: dismissedArtworks,
+      },
+    },
+    aggregations: {
+      uniq_artists: {
+        terms: {
+          field: "artist_id",
+          size: 1,
+          order: {
+            top_score: "desc",
+          },
+        },
+        aggs: {
+          sim_works: {
+            top_hits: {
+              size: 1,
+              _source: ["_id"],
+            },
+          },
+          top_score: {
+            max: {
+              script: "_score",
+            },
+          },
+        },
+      },
+    },
+  }
+}
+
+const CURATORS_PICKS_QUERY = `
+  query {
+    marketingCollection(slug: "curators-picks") {
+      slug
+      artworkIds
+    }
+  }
+`
+
+const ARTWORK_QUERY = internalID => `
+  query {
+    artwork(id: "${internalID}") {
+      title
+      imageUrl
+      internalID
+      artist {
+        name
+      }
+    }
+  }
+`
+
+const request = async (url, opts) => {
+  try {
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: opts.body,
+    }
+
+    const response = await fetch(url, options)
+
+    if (!response?.ok) {
+      throw new Error(`Response status: ${response.status}`)
+    }
+
+    const json = await response.json()
+    return json
+  } catch (error) {
+    console.log("Error fetching data")
+    console.error(error.message)
+  }
+}
+
+export const InfiniteDiscoveryApp = () => {
+  const [artworks, setArtworks] = React.useState([]) as any
+  const [likedArtworks, setLikedArtworks] = React.useState([]) as any
+  const [dismissedArtworks, setDismissedArtworks] = React.useState([]) as any
+  const [curatedArtworks, setCuratedArtworks] = React.useState([]) as any
+
+  const onLike = artwork => {
+    setLikedArtworks([...likedArtworks, artwork])
+    // filter out from curated artworks
+    setCuratedArtworks(curatedArtworks.filter(a => a !== artwork))
```

Comment:
Will we run into stale state issues by using curatedArtworks this way? Does it need to be an [updater function](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state)? So something like setCuratedArtworks(curatedArtworks => curatedArtworks.filter(...)

Same Q for the other `set...` invocations?